File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.java
import java.util.*;

class Solution {
    public int[] solution(String today, String[] terms, String[] privacies) {
        List<Integer> answer = new LinkedList<Integer>();
        
        String[] tmpToday = today.split("[.]");
        int TODAY = Integer.parseInt(tmpToday[0]) * 12 * 28 + Integer.parseInt(tmpToday[1]) * 28 + Integer.parseInt(tmpToday[2]);
        
        Map<String, Integer> EXP = new HashMap<String, Integer>();
        for (String term: terms) {
            String[] tmp = term.split(" ");
            EXP.put(tmp[0], Integer.parseInt(tmp[1]) * 28);
        }
        
        for (int i = 0; i < privacies.length; i++) {
            String[] privacy = privacies[i].split(" ");
            
            String[] tmpExp = privacy[0].split("[.]");
            int exp = Integer.parseInt(tmpExp[0]) * 12 * 28 + Integer.parseInt(tmpExp[1]) * 28 + Integer.parseInt(tmpExp[2]);
            
            String term = privacy[1];
            
            if(exp + EXP.get(term) <= TODAY) {
                answer.add(i+1);
            }
        }

        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
def solution(today, terms, privacies):
    answer = []

    # terms를 dictionary로 만듦
    td = {}
    for term in terms:
        tmpS, tmpI = term.split()
        td[tmpS] = int(tmpI)
    
    
    for n in range(len(privacies)):
        # 날짜와 약관을 분리, 날짜를 년월일 분리해서 숫자로
        tmpD, T = privacies[n].split()
        y, m, d = map(int, tmpD.split('.'))
        Y, M, D = map(int, today.split('.'))
        
        # 유효기간 구하기
        # 월
        exp = y * 12 * 28 + (m + td[T]) * 28 + d - 1       
        EXP = Y * 12 * 28 + M * 28 + D
        if exp < EXP:
            answer.append(n+1)
    return answer